cmds/rush/readline.go

```diff
@@ -0,0 +1,128 @@
+package main
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	"github.com/chzyer/readline"
+)
+
+func usage(w io.Writer) {
+	io.WriteString(w, "commands:\n")
+	io.WriteString(w, completer.Tree("    "))
+}
+
+// Function constructor - constructs new function for listing given directory
+func listFiles(path string) func(string) []string {
+	return func(line string) []string {
```

Comment:
Miss understand of the language, my bad. So the listFiles function is, in a way, returning a new function, it constructs a new function for a given directory right?